### PR TITLE
Consider edge cases for UTXO expiration, add tests. Closes #275

### LINKED ISSUE #275: Fix UTXO expiration
Complete with tests

### PR DESCRIPTION
**Pull request by pietro909**
I realized there is an edge case where `exitPaths` can be empty. It shouldn't happen but the case exists, therefore I decided for an execution path (no expiration) and blinded it with a test.

We can consider throwing an error instead if this is a _must never happen_ scenario.

## Summary by CodeRabbit

* **Refactor**
  * Improved UTXO expiration detection to determine boarding timelocks from multiple exit paths, using the earliest timelock and safely handling cases with no exit paths.

* **Tests**
  * Added unit tests covering confirmed/non-expired filtering, earliest-timelock selection, and behavior when exit paths are absent.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by louisinger — APPROVED** [review]
in fact, exit paths are optional, nice catch!

File: src/test/lib/utxo.test.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    exitPaths = []
+    const wallet = {
+      getBoardingUtxos: vi.fn().mockResolvedValue([confirmedNotExpired]),
+    } as any
+
+    const result = await getConfirmedAndNotExpiredUtxos(wallet)
+
+    expect(result).toEqual([confirmedNotExpired])
+    expect(hasBoardingTxExpired).toHaveBeenCalledTimes(0)
+  })
```
[the remark this answers is not in the document]
Comment (louisinger) [reply src/test/lib/utxo.test.ts]: @pietro909 maybe the rabbit is right ?
